squire_lib/src/ffi_tournament.rs

```diff
@@ -215,8 +255,16 @@ impl TournamentId {
 
     /// Closes a tournament removing it from the internal FFI state
     #[no_mangle]
-    pub extern "C" fn close_tourn(self: Self) {
-        FFI_TOURNAMENT_REGISTRY.get().unwrap().remove(&self);
+    pub unsafe extern "C" fn close_tourn(self: Self) -> bool {
+        match FFI_TOURNAMENT_REGISTRY.get_mut().unwrap().remove(&self) {
```

Comment:
You probably want to close the tournament with `TournOp::Close()` and remove the file.